solutions/src/main/java/ru/sberbank/school/task11/Streams.java

```diff
@@ -0,0 +1,184 @@
+package ru.sberbank.school.task11;
+
+import lombok.NonNull;
+
+import java.util.*;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+
+/**
+ * Облегченная версия класса {@link Stream}
+ */
+public class Streams<T> {
+    private final List<T> items;
+    private final LinkedList<Function<List<T>, List<?>>> functions = new LinkedList<>();
+
+    private Streams(@NonNull Collection<T> stream) {
+        this.items = new ArrayList<>(stream);
+    }
+
+    @SafeVarargs
+    private Streams(@NonNull T ... args) {
+        this.items = Arrays.asList(args);
+    }
+
+    /**
+     * Принимает на вход массив элементов и возвращает стрим построенный на основе этих элементов.
+     *
+     * @param elements входные элементы
+     * @return стрим элементов из elements
+     */
+    @SafeVarargs
+    public static <T> Streams<T> of(@NonNull T... elements) {
+        return new Streams<>(elements);
+    }
+
+    /**
+     * Принимает на вход коллекцию элементов и возвращает стрим построенный на основе этих элементов.
+     *
+     * @param elements входные элементы
+     * @return стрим элементов из elements
+     */
+    public static <T> Streams<T> of(@NonNull Collection<T> elements) {
+        return new Streams<>(elements);
+    }
+
+    /**
+     * Фильтрация элементов по заданному правилу. Необходимо подобрать нужый интерфейс для передачи в этот метод.
+     *
+     * <p>Иными словами, этот метод оставляет в коллекции только те элементы, которые удовлетворяют условию в лямбде.
+     *
+     * <p>Аналог Stream#map(Function).
+     *
+     * <p>Intermediate операция.
+     *
+     * @param predicate - правило фильтрации элементов.
+     * @return стрим
+     */
+    public Streams<T> filter(Predicate<? super T> predicate) {
+        functions.add((items) -> {
+            List<T> result = new ArrayList<>();
+
+            for (T item : items) {
+                if (predicate.test(item)) {
+                    result.add(item);
+                }
+            }
+
+            return result;
+        });
+
+        return this;
+    }
+
+    /**
+     * Преобразование элементов в какие-то другие элементы. Необходимо подобрать нужый интерфейс для передачи в этот метод.
```

Comment:
Line is longer than 120 characters (found 123).

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a1052066136>*